uptane-standard.md

```diff
@@ -385,24 +385,109 @@ The Snapshot role SHALL produce and sign metadata about all Targets metadata the
+
+* A release counter, to be incremented each time a new version of the image is released. This can be used to prevent rollback attacks even in cases where the director repository is compromised.
+* A hardware identifier, or list of hardware identifiers, representing models of ECU that the image is compatible with. This can be used to ensure that an ECU can't be ordered to install an incompatible image, even in cases where the director repository is compromised.
+
+The following information SHOULD be provided for each image on the director repository:
+
+* An ECU identifier, specifying (by serial number, for example) of the ECU that should install the image.
+* If encrypted images are desired, information about filenames, hashes, and file size of the encrypted image
+* If encrypted images are desired, information about the encryption method, and other relevant information--for example, a symmetric encryption key encrypted by the ECU's asymmetric key could be included in the director's metadata.
+
+A download URL for the image file MAY be provided by the director repository. This may be useful when the image is on a public CDN and the director wishes to provide a signed URL, or as an alternative to implementing the map file when working with encrypted images downloaded from the director.
 
 #### Metadata about Delegations {#delegations_meta}
 
+A targets metadata file on the image repository (but not the director repository) MAY delegate signing authority to other entities--for example, delegating signing authority for a particular ECU's firmware to that ECU's supplier. A metadata file MAY contain more than one delegation, and MUST keep the delegations in prioritized order.
+
+A list of delegations MUST provide the following information:
+
+* A list of public keys of all delegatees. Each key should have a unique identifier, and a key type.
+* A list of delegations, each of which contains:
+  * A list of the images or paths this role applies to. This MAY be expressed using wildcards, or by enumerating a list, or a combination of the two.
+  * An indicator of whether this is a terminating delegation or not. (See {{targets_role_delegations}}.)
+  * A list of the roles this delegation applies to. Each role needs to specify:
+    * A name for the role (e.g. "supplier1-qa")
+    * The key identifiers for each key this role uses
+    * A threshold of keys which must sign for this role
+
+Note that **any** targets metadata file may contain delegations--delegations can be in chains of arbitrary length.
+
 ### Snapshot Metadata {#snapshot_meta}
 
+The snapshot metadata lists version numbers and filenames of all targets metadata files. It protects against mix-and-match attacks in the case that a delegated supplier key has been compromised.
+
+For each targets metadata file on the repository, the snapshot metadata SHALL contain the following information:
+
+* The filename and version number of the each targets metadata file on the repository
```

Comment:
"The filename and version number of ~~**the**~~ each targets metadata file **in** the repository"